### ISSUE TITLE
FP32 causes 8GB vram gpu out of memory error, RunDiffusion/Juggernaut-X-v10 FP16 is not supported

### ISSUE THREAD

**Issue by xhoxye** [lllyasviel/Omost#63]
`# SDXL`
`sdxl_name = RunDiffusion/Juggernaut-X-v10 #FP16 is not supported`
`# sdxl_name = SG161222/RealVisXL_V4.0`
`# sdxl_name = stabilityai/stable-diffusion-xl-base-1.0`

![QQ截图20240604070222](https://github.com/lllyasviel/Omost/assets/129571231/5a263a95-04e3-4d37-b875-1a442bef7354)

FP32 can be downloaded, however it causes 8GB vram gpu out of memory error

**Comment by Duemellon** [comment]
I see the syntax you used to change models but I'm not sure how to apply that to my own. Of course all my models are located in the A1111 folder somewhere else. What is the syntax? Is there a problem concern with the file extension?

**Comment by xhoxye** [comment]
@Duemellon
It's not a .safetensors file name, it's the huggingface directory name
https://huggingface.co/SG161222/RealVisXL_V4.0

**Comment by Duemellon** [comment]
So you can't point this to a local download of a file at this time? It can only do a download after you point to the HF link?

**Comment by xhoxye** [comment]
If you want to load the local .safetensors file, you will need to modify more code

**Comment by Manul07** [comment]
no undesend

**Comment by Duemellon** [comment]
> If you want to load the local .safetensors file, you will need to modify more code

yes, that's what would be needed. I have a terabyte or more of locally installed safetensor fiels that I could use instead of pointing to a a HF directory. I'm not familiar with PY at all to make these changes but that is a different topic than this thread at this point.

**Comment by xhoxye** [comment]
models_dir = os.path.join(os.getcwd(), 'models/checkpoints')

    sdxl_name = 'RealVisXL_V4.0'

    model_path = os.path.join(models_dir, sdxl_name + '.safetensors')

    if os.path.isfile(model_path):
        temp_pipeline = StableDiffusionXLImg2ImgPipeline.from_single_file(model_path)
        
        tokenizer = temp_pipeline.tokenizer
        tokenizer_2 = temp_pipeline.tokenizer_2
        text_encoder = temp_pipeline.text_encoder
        text_encoder_2 = temp_pipeline.text_encoder_2
        text_encoder_2 = CLIPTextModel(config=text_encoder_2.config)
        vae = temp_pipeline.vae
        unet = temp_pipeline.unet
    else:
        raise FileNotFoundError(f"Model file {model_path} not found.")

**Comment by xhoxye** [comment]
![QQ截图20240606095725](https://github.com/lllyasviel/Omost/assets/129571231/77b08e8c-ac44-4f20-9a0e-effcafde08df)
![QQ截图20240606095718](https://github.com/lllyasviel/Omost/assets/129571231/680f6d55-361e-427a-83a0-41ee9fb400bf)
![QQ截图20240606095707](https://github.com/lllyasviel/Omost/assets/129571231/5de57552-9db6-4240-b966-515cfe63f1ec)
![QQ截图20240606095646](https://github.com/lllyasviel/Omost/assets/129571231/4452244a-3e60-41a0-b198-770520f69cda)

![QQ截图20240606095630](https://github.com/lllyasviel/Omost/assets/129571231/b335e658-c8ef-48bb-8545-ee221ea89f31)

**Comment by xhoxye** [comment]
I'll submit a PR later

**Comment by xhoxye** [comment]
"torch_dtype=torch.float32"  It run incorrectly

**Comment by xhoxye** [comment]
https://github.com/lllyasviel/Omost/pull/81
